go/chat/attachment_httpsrv.go

```diff
@@ -269,6 +294,98 @@ func (r *AttachmentHTTPSrv) serveUnfurlAsset(w http.ResponseWriter, req *http.Re
 	}
 }
 
+type giphyGalleryInfo struct {
+	results []chat1.GiphySearchResult
+	convID  chat1.ConversationID
+	tlfName string
+}
+
+func (r *AttachmentHTTPSrv) getGiphyGallerySelectURL(ctx context.Context, convID chat1.ConversationID,
+	tlfName, targetURL string) string {
+	addr, err := r.httpSrv.Addr()
+	if err != nil {
+		r.Debug(ctx, "getGiphySelectURL: failed to get HTTP server address: %s", err)
+		return ""
+	}
+	return fmt.Sprintf("http://%s/%s?url=%s&convID=%s&tlfName=%s", addr, r.giphySelectEndpoint,
+		url.QueryEscape(targetURL), convID, tlfName)
+}
+
+func (r *AttachmentHTTPSrv) serveGiphyGallerySelect(w http.ResponseWriter, req *http.Request) {
+	ctx := globals.ChatCtx(context.Background(), r.G(), keybase1.TLFIdentifyBehavior_CHAT_GUI, nil,
+		NewSimpleIdentifyNotifier(r.G()))
+	defer r.Trace(ctx, func() error { return nil }, "serveGiphyGallerySelect")()
+	url := req.URL.Query().Get("url")
+	strConvID := req.URL.Query().Get("convID")
+	tlfName := req.URL.Query().Get("tlfName")
+	convID, err := chat1.MakeConvID(strConvID)
+	if err != nil {
+		r.makeError(context.TODO(), w, http.StatusInternalServerError, "failed to decode convID: %s",
+			err)
+		return
+	}
+	if err := r.G().ChatHelper.SendTextByID(ctx, convID, tlfName, url); err != nil {
+		r.makeError(context.TODO(), w, http.StatusInternalServerError, "failed to send giphy url: %s",
+			err)
+	}
+	ui, err := r.G().UIRouter.GetChatUI()
+	if err == nil && ui != nil {
+		ui.ChatGiphyToggleResultWindow(ctx, convID, false, true)
+	}
+}
+
+func (r *AttachmentHTTPSrv) serveGiphyGallery(w http.ResponseWriter, req *http.Request) {
+	ctx := globals.ChatCtx(context.Background(), r.G(), keybase1.TLFIdentifyBehavior_CHAT_GUI, nil,
+		NewSimpleIdentifyNotifier(r.G()))
+	defer r.Trace(ctx, func() error { return nil }, "serveGiphyGallery")()
+	key := req.URL.Query().Get("key")
+	divStyle, err := url.QueryUnescape(req.URL.Query().Get("divstyle"))
+	if err != nil {
+		r.Debug(ctx, "serveGiphyGallery: failed to decode divstyle: %s", err)
```

Comment:
~should `divStyle`/`vidStyle` be set to `""` or some sensible default if there is an error ?~ see comment below